.github/workflows/pr_qc.yml

```diff
@@ -0,0 +1,21 @@
+name: Pull Request Quality Checks
+on:
+  pull_request:
+    types: [ opened, synchronize ]
+    branches: [ main ]
+jobs:
+  run-qa:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v4
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.11
+      - name: Run qa
+        env:
+            OPENAI_API_KEY: ${{ secrets.OPENAI_API_KEY }}
+        run: |
+          pip install ".[dev]"
```

Comment:
we can just do `make dev` instead of the full pip command (along with `make qa`) here for simplicity